File: 2024/day04-1.py
# Puntuacíon: 4 estrellas

def createXmasTree(height, ornament):
  result = ''

  for i in range(height):
    result += ['_'*(height-i-1)] + [ornament*(i*2+1)] + ['_'*(height-i-1)] + '\n'

  base = ['_'*(height-1)] + '#' + ['_'*(height-1)]

  result += base + '\n' + base

  return result

File: 2024/day04-1.js
// Puntuación: 5 estrellas

function createXmasTree(height, ornament) {
  let result = ''

  for(let i=0;i<height;i++) {
    result += '_'.repeat(height-i-1) + ornament.repeat(i*2+1) + '_'.repeat(height-i-1) + '\n'
  }

  let base = '_'.repeat(height-1) + '#' + '_'.repeat(height-1)

  result += base + '\n' + base

  return result
}
